vision-quickstart/app/src/main/java/com/google/mlkit/vision/demo/VisionProcessorBase.java

```diff
@@ -88,8 +89,19 @@
     private ByteBuffer processingImage;
     @GuardedBy("this")
     private FrameMetadata processingMetaData;
-    private ArrayList<Pair<Bitmap, Long>> bufferImageList = new ArrayList<>();
-    private Integer depth = 30;
+
+    class bufferImage
+    {
+        Bitmap image;
+        ImageInfo info;
+        bufferImage(Bitmap image, ImageInfo info)
+        {
+            this.image = image;
+            this.info = info;
+        }
+    };
+    private ArrayList<bufferImage> bufferImageList = new ArrayList<>();
```

Comment:
Initialize the array list with the correct capacity, since you know it in advance.